TACC/Core-Styles
# chore: 0.5.2-c → 0.7.0 (migrate from tup-ui)

## Overview / Changes

- Revive Core-Styles.
- Back-port [TACC/tup-ui:/libs/core-styles](https://github.com/TACC/tup-ui/tree/main/libs/core-styles).
- Upgrade to v0.7.0.

## Related

Upcoming PRs that build off this migration/revival.

## Testing / Screenshots

This v0.7.0 has already been tested on clients.

To test locally, follow these guides:
- https://github.com/TACC/Core-Styles/wiki/How-to-Test-Version-Change \*
- https://github.com/TACC/Core-Styles/wiki/How-to-Test-Build-Output †

\* File count should decrease from `66` to `63`.
<details><summary>† Expect mostly version update in comment, but also some other changes. <sub>(click to toggle details)</sub></summary>

- version in comment from `0.5.2-c+` to `0.7.0-c+`
- `bootstrap.figure.css` does not have tags in selectors
- `c-button.css` has minor changes
- `cortal.icon.…` removed (I will restore them after migration, they were not in tup-ui yet)
- `figure.css` removed
- `html-elements.css` has minor changes
- `o-section.css` has some changes (at middle and bottom of file)
- `o-site.css` is empty (besides comment)
- `colors.css` has more variables
- `space.css` has more variables and `px` not `rem`

</details>

## Notes

<details><summary>Updating this repo speeds CMS development until tup-ui architecture is ready.</summary>

Architecture questions affecting @tacc/core-styles remain unanswered for [TACC/tup-ui](https://github.com/TACC/tup-ui/)—specifically, whether to bring in [TACC/Core-CMS](https://github.com/TACC/Core-CMS) or freeze [TACC/tup-ui:/libs/core-styles](https://github.com/TACC/tup-ui/tree/main/libs/core-styles) or take out [TACC/tup-ui:/libs/core-styles](https://github.com/TACC/tup-ui/tree/main/libs/core-styles).

The delay slows my CMS work. My speed of development exceeds the speed for which [TACC/tup-ui](https://github.com/TACC/tup-ui/) is ready. I have [Core-CMS—Core-Styles co-development solutions](https://github.com/TACC/Core-CMS/wiki/Locally-Develop-CMS---Styles) that works regardless where @tacc/core-styles is developed.

Having @tacc/core-styles in an independent repo allows PRs makes testing more reliable, because the client repo can point to a Git tag, branch, or commit.

If `core-styles/` is inside a monorepo, client repo can not pin like that unless using a third-party like https://gitpkg.now.sh/ (bothers [Core-Portal](https://github.com/TACC/Core-Portal) lead dev) **or** an NPM version of @tacc/core-styles (bothers [TACC/tup-ui](https://github.com/TACC/tup-ui/) lead dev).

When/If @tacc/core-styles will be developed in [TACC/tup-ui](https://github.com/TACC/tup-ui/), I will migrate @tacc/core-styles back to there.

</details>
